File: modules/fbs-core/api/src/main/scala/de/thm/ii/fbs/model/CheckrunnerConfiguration.scala
package de.thm.ii.fbs.model

import com.fasterxml.jackson.annotation.{JsonIgnore, JsonProperty}

/**
 * Checker configuration
 *
 * @param checkerType           The checker type, identifies the checkrunner
 * @param ord                   The checkrunner position in the order of checkrunner for a task
 * @param mainFileUploaded      True if the main file for a checker was uploaded
 * @param secondaryFileUploaded True if the secondary file for a checker was uploaded
 * @param id                    The configuration id, if 0, then an id was not assigned by the system
 */
case class CheckrunnerConfiguration(@JsonProperty checkerType: String,
                                    @JsonProperty ord: Int,
                                    @JsonProperty mainFileUploaded: Boolean = false,
                                    @JsonProperty secondaryFileUploaded: Boolean = false,
                                    @JsonIgnore id: Int = 0,
                                    @JsonIgnore taskId: Int = 0,
                                    @JsonProperty checkerTypeInformation: Option[CheckerTypeInformation] = None,
                                    @JsonProperty isInBlockStorage: Boolean = false
)

File: modules/fbs-core/api/src/main/kotlin/de/thm/ii/fbs/model/v2/CheckrunnerConfiguration.kt
package de.thm.ii.fbs.model.v2

import javax.json.Json
import javax.persistence.*

@Entity
@Table(name = "checkrunner_configuration")
class CheckrunnerConfiguration(
        @Id
        @Column(nullable = false, name = "configuration_id")
        var id: Int,
        @Column(nullable = false, name = "task_id")
        var taskId: Int,

        @Column(nullable = false, name = "checker_type")
        var checkerType: String,

        @Column(nullable = false, name = "main_file_uploaded")
        var mainFileUploaded: Boolean = false,

        @Column(nullable = false, name = "secondary_file_uploaded")
        var secondaryFileUploaded: Boolean = false,

        @Column(nullable = false)
        var ord: Int,

        @Column(name = "checker_type_information")
        var checkerTypeInformation: Json? = null, // TODO? json as ??

        @Column(nullable = false, name = "is_in_block_storage")
        var isInBlockStorage: Boolean = false,
)